be/src/exprs/string_functions.cpp

```diff
@@ -1746,25 +1748,42 @@ static inline const char* skip_leading_spaces(const char* begin, const char* end
         }
     }
 #endif
-    for (; p < end && *p == ' '; ++p) {
+    if constexpr (trim_single) {
+        if (remove[0] == ' ') {
+            for (; p < end && *p == ' '; ++p) {
+            }
+        } else {
+            for (; p < end && *p == remove[0]; ++p) {
+            }
+        }
+    } else if constexpr (!trim_utf8) {
+        for (; p < end && remove.find(*p) != remove.npos; ++p) {
+        }
+    } else {
+        size_t char_size = UTF8_BYTE_LENGTH_TABLE[(uint8_t)*p];
+        while (p < end && utf8_contains(remove, utf8_index, {p, char_size})) {
+            char_size = UTF8_BYTE_LENGTH_TABLE[(uint8_t)*p];
+            p += char_size;
+        }
     }
     return p;
 }
 
 // strings with little spaces can not benefit from simd optimization,
 // simd_optimization = false, turn off SIMD optimization
-template <bool simd_optimization>
-static const char* skip_trailing_spaces(const char* begin, const char* end) {
+template <bool simd_optimization, bool trim_single, bool trim_utf8>
+static const char* skip_trailing_spaces(const char* begin, const char* end, const std::string& remove,
```

Comment:
Use uint64_t/int64_t instead of std::string&
uint64_t/int64_t can here can be saved in register directly,
but for accessing elements of std::string&,  it must be avoiding loading it from the memory.uint64_t is wide enougth to hold ascii char or utf8 char.

Reply:
well, I think you have some misunderstanding over this `remove` parameter, it represents several characters, instead of single. So it should be a string.

An example is, `trim(str, 'abcde')`, in this case the `remove` is `abcde`, which is a a string.